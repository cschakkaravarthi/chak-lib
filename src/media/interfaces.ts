export interface SourceMediaProps {
  title: string;
  bodyText?: string;
  subsection?: string;
  isFetching?: boolean;
  sectionImageSrc?: string;
}
