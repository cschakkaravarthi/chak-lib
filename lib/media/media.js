"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var mediaPrimary_1 = require("./mediaPrimary");
var mediaSecondary_1 = require("./mediaSecondary");
exports.Media = function (props) {
    var _a = props.variant, variant = _a === void 0 ? 'primary' : _a;
    switch (variant) {
        case 'primary':
            return react_1["default"].createElement(mediaPrimary_1.MediaPrimary, __assign({}, props));
        case 'secondary':
            return react_1["default"].createElement(mediaSecondary_1.MediaSecondary, __assign({}, props));
    }
    return null;
};
//# sourceMappingURL=media.js.map