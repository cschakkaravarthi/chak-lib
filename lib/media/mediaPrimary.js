"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var react_bootstrap_1 = require("react-bootstrap");
require("../../scss/_media.scss");
exports.MediaPrimary = function (props) {
    var title = props.title, bodyText = props.bodyText, sectionImageSrc = props.sectionImageSrc;
    return (react_1["default"].createElement(react_bootstrap_1.Media, { bsPrefix: "media" },
        react_1["default"].createElement("img", { width: 64, height: 64, alt: title, src: sectionImageSrc, className: 'mr-4 rounded-circle mt-3' }),
        react_1["default"].createElement(react_bootstrap_1.Media.Body, null,
            react_1["default"].createElement("p", { className: 'font-weight-bold font-size-lg' }, title),
            react_1["default"].createElement("p", { className: 'font-size-base' }, bodyText))));
};
//# sourceMappingURL=mediaPrimary.js.map