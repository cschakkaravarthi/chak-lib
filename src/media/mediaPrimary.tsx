import React, { FC } from 'react';
import { Media as SourceMedia } from 'react-bootstrap';
import Skeleton from 'react-loading-skeleton';

import { SourceMediaProps } from './interfaces';

import '../../scss/_media.scss';

export const MediaPrimary: FC<SourceMediaProps> = props => {
  const { title, bodyText, sectionImageSrc, isFetching = false } = props;

  return (
    <SourceMedia bsPrefix="media">
      {!isFetching ? (
        <img width={64} height={64} alt={title} src={sectionImageSrc} className="mr-4 rounded-circle mt-3" />
      ) : (
        <span className="circle-skeleton mr-4 mt-3">
          <Skeleton circle={true} height={64} width={64} />
        </span>
      )}
      <SourceMedia.Body>
        <p className="font-weight-bold font-size-lg">{!isFetching ? title : <Skeleton width={150} />}</p>
        <p className="font-size-base">{!isFetching ? bodyText : <Skeleton width={300} />}</p>
      </SourceMedia.Body>
    </SourceMedia>
  );
};
