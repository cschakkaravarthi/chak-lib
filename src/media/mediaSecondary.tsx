import React, { FC } from 'react';
import { Media as SourceMedia } from 'react-bootstrap';
import { SourceMediaProps } from './interfaces';
import Skeleton from 'react-loading-skeleton';

import '../../scss/_media.scss';
import { images } from '../assets/images';

export const MediaSecondary: FC<SourceMediaProps> = props => {
  const { title, subsection, sectionImageSrc, isFetching = false } = props;

  return (
    <SourceMedia bsPrefix="media">
      {!isFetching ? (
        <img width={64} height={64} src={sectionImageSrc} alt="Generic placeholder" className="mr-4 rounded-circle" />
      ) : (
        <span className="circle-skeleton mr-4">
          <Skeleton circle={true} height={64} width={64} />
        </span>
      )}
      <SourceMedia.Body>
        <div className="font-size-lg mt-3">
          {!isFetching ? title : <Skeleton width={100} />}
          <span className="mx-3">
            {!isFetching ? (
              <img src={images.caretRight} width={10} height={30} className="gray-600 d-inline" />
            ) : (
              <Skeleton width={15} />
            )}
          </span>
          <span className="font-weight-bold">{!isFetching ? subsection : <Skeleton width={150} />}</span>
        </div>
      </SourceMedia.Body>
    </SourceMedia>
  );
};
