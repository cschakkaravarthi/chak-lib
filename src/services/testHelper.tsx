/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.

import ShallowRenderer from 'react-test-renderer/shallow';

export const shallowRender = (toRender: any): Record<string, any> => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};
