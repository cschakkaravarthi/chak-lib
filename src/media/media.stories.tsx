import React from 'react';
import StoriesContainer from '../StoriesContainer';
import { Media } from './media';

export default {
  component: Media,
  title: 'Media'
};

export const mediaPrimary = (): JSX.Element => (
  <StoriesContainer>
    <Media
      variant="primary"
      title="Media title primary"
      sectionImageSrc="https://picsum.photos/id/522/200/200"
      bodyText="Some text to see how the primary body text looks"
    />

    <hr />

    <Media
      isFetching={true}
      variant="primary"
      title="Media title primary"
      sectionImageSrc="https://picsum.photos/id/522/200/200"
      bodyText="Some text to see how the primary body text looks"
    />
  </StoriesContainer>
);

export const mediaSecondary = (): JSX.Element => (
  <StoriesContainer>
    <Media
      title="Section"
      variant="secondary"
      subsection="Subsection"
      sectionImageSrc="https://picsum.photos/id/522/200/200"
    />

    <hr />

    <Media
      title="Section"
      isFetching={true}
      variant="secondary"
      subsection="Subsection"
      sectionImageSrc="https://picsum.photos/id/522/200/200"
    />
  </StoriesContainer>
);
