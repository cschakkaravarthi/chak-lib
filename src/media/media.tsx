import React, { FC } from 'react';
import { SourceMediaProps } from './interfaces';
import { MediaPrimary } from './mediaPrimary';
import { MediaSecondary } from './mediaSecondary';

export interface MediaProps extends SourceMediaProps {
  variant?: string;
}

export const Media: FC<MediaProps> = props => {
  const { variant = 'primary' } = props;

  switch (variant) {
    case 'primary':
      return <MediaPrimary {...props} />;
    case 'secondary':
      return <MediaSecondary {...props} />;
  }

  return null;
};
