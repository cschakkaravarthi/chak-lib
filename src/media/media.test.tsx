import React from 'react';
import { shallowRender } from '../services/testHelper';
import { Media } from './media';
import { MediaPrimary } from './mediaPrimary';
import { MediaSecondary } from './mediaSecondary';

describe('Media', () => {
  it('should render primary variant correctly', () => {
    const tree = shallowRender(
      <Media
        variant="primary"
        title="Media title primary"
        sectionImageSrc="https://picsum.photos/id/522/200/200"
        bodyText="Some text to see how the primary body text looks"
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render secondary variant correctly', () => {
    const tree = shallowRender(
      <Media
        title="Section"
        variant="secondary"
        subsection="Subsection"
        sectionImageSrc="https://picsum.photos/id/522/200/200"
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render primary correctly', () => {
    const tree = shallowRender(
      <MediaPrimary
        title="Media title primary"
        sectionImageSrc="https://picsum.photos/id/522/200/200"
        bodyText="Some text to see how the primary body text looks"
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render secondary correctly', () => {
    const tree = shallowRender(
      <MediaSecondary
        title="Section"
        subsection="Subsection"
        sectionImageSrc="https://picsum.photos/id/522/200/200"
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
