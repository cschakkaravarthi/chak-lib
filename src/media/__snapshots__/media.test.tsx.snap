// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media should render primary correctly 1`] = `
<ForwardRef
  bsPrefix="media"
>
  <img
    alt="Media title primary"
    className="mr-4 rounded-circle mt-3"
    height={64}
    src="https://picsum.photos/id/522/200/200"
    width={64}
  />
  <ForwardRef>
    <p
      className="font-weight-bold font-size-lg"
    >
      Media title primary
    </p>
    <p
      className="font-size-base"
    >
      Some text to see how the primary body text looks
    </p>
  </ForwardRef>
</ForwardRef>
`;

exports[`Media should render primary variant correctly 1`] = `
<Unknown
  bodyText="Some text to see how the primary body text looks"
  sectionImageSrc="https://picsum.photos/id/522/200/200"
  title="Media title primary"
  variant="primary"
/>
`;

exports[`Media should render secondary correctly 1`] = `
<ForwardRef
  bsPrefix="media"
>
  <img
    alt="Generic placeholder"
    className="mr-4 rounded-circle"
    height={64}
    src="https://picsum.photos/id/522/200/200"
    width={64}
  />
  <ForwardRef>
    <div
      className="font-size-lg mt-3"
    >
      Section
      <span
        className="mx-3"
      >
        <img
          className="gray-600 d-inline"
          height={30}
          src="test-file-stub"
          width={10}
        />
      </span>
      <span
        className="font-weight-bold"
      >
        Subsection
      </span>
    </div>
  </ForwardRef>
</ForwardRef>
`;

exports[`Media should render secondary variant correctly 1`] = `
<Unknown
  sectionImageSrc="https://picsum.photos/id/522/200/200"
  subsection="Subsection"
  title="Section"
  variant="secondary"
/>
`;
